src/compiler/checker.ts

```diff
@@ -22814,11 +22814,7 @@ namespace ts {
             addDiagnostic(UnusedKind.Local, createDiagnosticForNode(node, message, name));
         }
 
-        function parameterNameStartsWithUnderscore(parameterName: DeclarationName) {
-            return parameterName && isIdentifierThatStartsWithUnderScore(parameterName);
-        }
-
-        function isIdentifierThatStartsWithUnderScore(node: Node) {
+        function isIdentifierThatStartsWithUnderscore(node: Node) {

```

Comment:
I also eliminated the now-redundant `parameterNameStartsWithUnderscore` and fixed the capitalisation of `isIdentifierThatStartsWithUnderscore`.